Add freiheit93 machine

mbriand opened the pull request:
This PR adds support for a new machine: the freiheit 93, based on the FRDM i.MX93 board.

It also switches all layers to the walnascar branch: maybe we don't want to merge this on master but on a new walnascar branch?

Main components used by the machine:
- U-Boot 2025.01: mainline, with a backported patch to add support of the machine.
- Linux 6.12: mainline, with a patch adding the machine device tree. I plan to submit this patch upstream in the coming days.
- tf-a: mainline.
- imx-boot-firmware-files: I imported the recipe from meta-freescale, and simplified it a bit. I added a `LICENSE_FLAGS = "NXP_EULA_v58"`, so user will have to read the license file and add `LICENSE_FLAGS_ACCEPTED += "NXP_EULA_v58"` in their local.conf.

Build has been tested with all of the three machines. I've only booted the images built for the freiheit93 and stompduck machines, as I don't have a bbb available here.

mbriand committed:
kas: Fix deprecated syntax

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand committed:
meta-kiss: bump to scarthgap

Switch to scarthgap branches and remove the no longer needed
trusted-firmware-a patch.

Also remove debug-tweaks: while this is still supported in scarthgap, it
is going to be removed in the next LTS. Replace it with single tweaks,
except for post-install-logging as this is not required for most use
cases.

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand committed:
linux-kiss: Add common include file

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand commented:
Switched from walnascar to scarthgap, everything else remains the same. I kept my walnascar branch, as it does contain fixes that may be useful later: mathieu/freiheit93-walnascar.

I ran the same tests, so I still need to test on bbb.

mbriand committed:
freiheit93: Enable fastboot mode in u-boot

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand committed:
distro: kiss: Use pni-names distro feature

Use persistent network interfaces names, avoiding interface names
changes across reboots.

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

lucaceresoli reviewed (commented):
Minor comment: you're not adding scarthgap but rather replacing kirkstone woth scarthgap (which is fine), so the subject line should be more like `meta-kiss: bump to scarthgap`.

lucaceresoli commented on .config.yaml:
```diff
@@ -34,4 +34,7 @@ local_conf_header:
+    # EXTRA_IMAGE_FEATURES += "empty-root-password \
+    #                          allow-empty-password \
+    #                          allow-root-login \
+    #                          "
```
AFAIK the complete replacement for `debug-tweaks` should include `post-install-logging`.
(https://docs.yoctoproject.org/migration-guides/migration-5.2.html)

I guess `post-install-logging` is not very important, so maybe add a note in the commit message like "replace debug-tweaks with the single tweaks it includes, except post-install-logging because ...".

Also you may mention in the commit message that migrating `debug-tweaks` is needed for a post-scarthgap version (walnascar) but we prefer being  future-proof.

mbriand replied:
OK, I'm going to add a bit of details in the commit message.

lucaceresoli reviewed (changes requested):
Overall good, but I added various comments.
Commits without comments LGTM.

kmaincent committed:
bsp: u-boot-kiss: Upgrade to v2025.07 and fix build dependencies

u-boot-common.inc is adding several CVE patches in the SRC_URI which are not
present in our u-boot-kiss recipe. Override SRC_URI to fix that and use the
official U-Boot repository with the latest release.

Fix build dependencies:
- Remove redundant python3-setuptools-native (inherited from
  u-boot-common.inc)
- Add missing gnutls-native dependency

Signed-off-by: Kory Maincent <kory.maincent@bootlin.com>
Co-developed-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>
Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand commented:
> Minor comment: you're not adding scarthgap but rather replacing kirkstone woth scarthgap (which is fine), so the subject line should be more like `meta-kiss: bump to scarthgap`.

Yes, makes sense. I had a separate branch in mind at first, but even then this was not a great title.

mbriand committed:
freiheit93: Add support for a new machine

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand committed:
linux: Add recipe for linux 6.16

Signed-off-by: Mathieu Dubois-Briand <mathieu.dubois-briand@bootlin.com>

mbriand commented on meta-kiss/recipes-bsp/u-boot/u-boot-kiss/0001-Revert-board-st-stm32mp1-Clean-env_get_location.patch:
@kmaincent Any insight about what is the issue here? I can update the u-boot recipe with the comment and update the PR, but I believe it would better if you can provide what would be the correct comment here.

kmaincent replied:
Here is the commit message of the patch sent mainline:
```
Restore support for environment storage in EXT4 filesystem on eMMC boot.
The previous cleanup incorrectly removed this fallback option which is
needed for boards that store their environment in an EXT4 partition.

This configuration is OS-specific rather than board-dependent and should
remain as it is configurable via menuconfig. Even if it is not described
in ST defconfigs people may have enabled it in their defconfig.
```
